app/store/store.js

```diff
@@ -46,180 +43,132 @@ const setTransforms = [
+            if (!channelViewBlackList[channelKey]) {
+                channel[channelKey] = inboundState.channel[channelKey];
             }
-
-            return inboundState;
-        },
-        (outboundState, key) => {
-            if (key === 'entities') {
-                const state = {...outboundState};
-                for (const prop in state) {
-                    if (state.hasOwnProperty(prop)) {
-                        state[prop] = transformSet(state[prop], setTransforms, false);
-                    }
+        }
+
+        return {
+            ...inboundState,
+            channel,
+        };
+    },
+    null,
+    {whitelist: ['views']}, // Only run this filter on the views state (or any other entry that ends up being named views)
+);
+
+const emojiBlackList = {nonExistentEmoji: true};
+const emojiBlackListFilter = createTransform(
+    (inboundState) => {
+        const emojis = {};
+
+        for (const emojiKey of Object.keys(inboundState.emojis)) {
+            if (!emojiBlackList[emojiKey]) {
+                emojis[emojiKey] = inboundState.emojis[emojiKey];
+            }
+        }
+
+        return {
+            ...inboundState,
+            emojis,
+        };
+    },
+    null,
+    {whitelist: ['entities']}, // Only run this filter on the entities state (or any other entry that ends up being named entities)
+);
+
+const setTransformer = createTransform(
+    (inboundState, key) => {
+        if (key === 'entities') {
+            const state = {...inboundState};
+            for (const prop in state) {
+                if (state.hasOwnProperty(prop)) {
+                    state[prop] = transformSet(state[prop], setTransforms);
                 }
-
-                return state;
             }
 
-            return outboundState;
-        },
-    );
-
-    const offlineOptions = {
-        effect: (effect, action) => {
-            if (typeof effect !== 'function') {
-                throw new Error('Offline Action: effect must be a function.');
-            } else if (!action.meta.offline.commit) {
-                throw new Error('Offline Action: commit action must be present.');
+            return state;
+        }
+
+        return inboundState;
+    },
+    (outboundState, key) => {
+        if (key === 'entities') {
+            const state = {...outboundState};
+            for (const prop in state) {
+                if (state.hasOwnProperty(prop)) {
+                    state[prop] = transformSet(state[prop], setTransforms, false);
+                }
             }
 
-            return effect();
-        },
-        persist: (store, options) => {
-            const persistor = persistStore(store, {storage: AsyncStorage, ...options}, () => {
-                store.dispatch({
-                    type: General.STORE_REHYDRATION_COMPLETE,
-                });
+            return state;
+        }
+
+        return outboundState;
+    },
+);
+
+export const persistConfig = {
```

Comment:
instead of exporting you could pass `persistConfig` as a parameter to `middlewares` like `...middlewares(persistConfig)` and then make the necessary changes to the middleware file.

Reply:
Done in ab0ebbdbe01d4fdd0100a2b6339f4d19a2045728